### PR TITLE
STY: x.__class__ to type(x) #batch-2

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref https://github.com/pandas-dev/pandas/pull/29816
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -948,7 +948,7 @@ def __repr__(self):
         """
         Return a string representation for this object.
         """
-        klass = self.__class__.__name__
+        klass = type(self).__name__
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py]: can you call this klass_name

Comment (MomIsBestFriend) [reply pandas/core/indexes/base.py]: sure:)

**Review by simonjayhawkins — APPROVED** [review]

**Comment by jbrockmendel** [comment]
LGTM

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

**Pull request merged**